src/pq/connection.cr

```diff
@@ -188,45 +189,44 @@ module PQ
       end
     end
 
+    def start_replication_frame_loop(publication_name : String, slot_name : String, &block : PG::Replication::Frame ->)
+      command = "START_REPLICATION SLOT #{slot_name} LOGICAL 0/0 (proto_version '1', binary 'true', publication_names '#{publication_name}')"
```

Comment:
A few notes on this line:

- There are currently [4 `proto_version` values](https://www.postgresql.org/docs/current/protocol-logical-replication.html#PROTOCOL-LOGICAL-REPLICATION-PARAMS), but anything other than `1` requires Postgres 14+ and changes how some `WALMessage`s are decoded. Supporting these dynamically would be more complicated, so I'm just supporting the baseline out of the gate.
- This is currently hardcoded to start at the beginning of the WAL (`0/0`). The `PG.listen_replication` method should almost certainly be updated to allow passing in other LSNs to start from a different point in the WAL history.

Reply:
Nice. Main addition of in v2 is transaction ids on various actions, which could be useful for some that might want this functionality. I wouldn't say it's necessary since the events are linear (all CRUD actions map to the transaction id from the most recent `Begin` WAL message), but it might still be useful.